src/devtools/store.js

```diff
@@ -862,15 +942,29 @@ export default class Store extends EventEmitter {
           key = ` key="${element.key}"`;
         }
 
+        let suffix = '';
+        if (includeWeight) {
+          suffix = ` (${element.isCollapsed ? 1 : element.weight})`;
+        }
+
         snapshotLines.push(
           `${'  '.repeat(element.depth + 1)}${prefix} <${element.displayName ||
-            'null'}${key}>`
+            'null'}${key}>${suffix}`
         );
       }
 
       rootWeight += weight;
     });
 
+    // Make sure the pretty-printed test align with the Store's reported number of total rows.
+    if (rootWeight !== this._weightAcrossRoots) {
+      throw Error(
```

Comment:
Nice

Reply:
Yeah, this was my bad. This was masking some bugs our earlier tests would have shown.